File: src/main/java/forpleuvoir/hiirosakura/client/feature/chatmessage/ChatMessageInject.java
package forpleuvoir.hiirosakura.client.feature.chatmessage;

import forpleuvoir.hiirosakura.client.config.Configs;

import java.util.regex.Pattern;

import static forpleuvoir.hiirosakura.client.config.Configs.Toggles.*;
import static forpleuvoir.hiirosakura.client.config.Configs.Values.*;

/**
 * 聊天消息注入
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.feature.chatmessage
 * <p>#class_name ChatMessageInject
 * <p>#create_time 2021/6/24 22:16
 */
public class ChatMessageInject {
    public static ChatMessageInject INSTANCE = new ChatMessageInject();

    public String handlerMessage(String message) {
        if (message.startsWith("/")) return message;
        //添加前缀以及后缀
        if (needInject(message))
            message = setFix(message);
        return message;
    }

    private boolean needInject(String message) {
        boolean returnValue = true;
        if (ENABLE_CHAT_MESSAGE_INJECT_REGEX.getBooleanValue()) {
            boolean matched = false;
            for (String regex : CHAT_MESSAGE_INJECT_REGEX.getStrings()) {
                if (Pattern.matches(regex, message)) {
                    matched = true;
                }
            }
            if (REVERSE_CHAT_MESSAGE_INJECT_REGEX.getBooleanValue()) matched = !matched;
            returnValue = !matched;
        }
        return returnValue;
    }

    private String setFix(String message) {
        if (Configs.Toggles.CHAT_MESSAGE_INJECT.getBooleanValue())
            return String.format("%s%s%s", CHAT_MESSAGE_INJECT_PREFIX.getStringValue(), message,
                                 CHAT_MESSAGE_INJECT_SUFFIX.getStringValue()
            );
        return message;
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/feature/chatmessage/ChatMessageInject.kt
package forpleuvoir.hiirosakura.client.feature.chatmessage

import forpleuvoir.hiirosakura.client.config.Configs.Toggles.CHAT_MESSAGE_INJECT
import forpleuvoir.hiirosakura.client.config.Configs.Toggles.ENABLE_CHAT_MESSAGE_INJECT_REGEX
import forpleuvoir.hiirosakura.client.config.Configs.Toggles.REVERSE_CHAT_MESSAGE_INJECT_REGEX
import forpleuvoir.hiirosakura.client.config.Configs.Values.CHAT_MESSAGE_INJECT_EXP
import forpleuvoir.hiirosakura.client.config.Configs.Values.CHAT_MESSAGE_INJECT_FILTER
import java.util.regex.Pattern

/**
 * 聊天消息注入
 *
 * @author forpleuvoir
 *
 * 项目名 hiirosakura
 *
 * 包名 forpleuvoir.hiirosakura.client.feature.chatmessage
 *
 * 文件名 ChatMessageInject
 *
 * 创建时间 2021/6/24 22:16
 */
object ChatMessageInject {
	private const val originMsg = "#{message}"

	@JvmStatic
	fun handlerMessage(message: String): String {
		var msg = message
		if (msg.startsWith("/")) return msg
		//添加前缀以及后缀
		if (needInject(msg)) msg = setFix(msg)
		return msg
	}

	private fun needInject(message: String): Boolean {
		var returnValue = true
		if (ENABLE_CHAT_MESSAGE_INJECT_REGEX.getValue()) {
			var matched = false
			for (regex in CHAT_MESSAGE_INJECT_FILTER.getValue()) {
				if (Pattern.matches(regex, message)) {
					matched = true
				}
			}
			if (REVERSE_CHAT_MESSAGE_INJECT_REGEX.getValue()) matched = !matched
			returnValue = !matched
		}
		return returnValue
	}

	private fun setFix(message: String): String {
		return if (CHAT_MESSAGE_INJECT.getValue()) {
			CHAT_MESSAGE_INJECT_EXP.getValue().replace(originMsg, message)
		} else message
	}

}